### DISCUSSION TITLE
need help setting up github on server

### DISCUSSION THREAD

**Discussion by ChaseBacon** [community/community#27970] [Actions]
So I am working on a FiveM server, I have github on my local PC and when i make changes I push them to the repo in my git account. The thing I am wanting to do is. When I push a commit, it automatically updates the files on my server. I have watched countless videos and tried to read countless threads on how to set this up and for the life of me can't figure it out. The way I'm doing it for the time being is having github desktop on both local maching and remote machine, push to repo, then pull repo changes on remote, then copy and past those files into the correct places in the FiveM server files. I would rather this happen automatically.   PLEASE HELP

**Comment by airtower-luna** [comment]
> When I push a commit, it automatically updates the files on my server.

You'll have to set up *something* to pull and deploy updates to your server. Some ways that could be achieved are:

* Set up a [webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks). You'll get HTTPS requests to a URL you configure for repository events (e.g. a push) you select.
* Set up an [Actions workflow](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) that calls a deployment tool (e.g. Ansible) to deploy the files to your server.
* Set up a cron job on your server that occasionally checks if there are new commits, and deploys if yes. This is slower and less elegant than the other two, but works without incoming connections.
